### ISSUE TITLE
Hierarchical RL

### ISSUE THREAD

**Issue by kevin-xuan** [alirezakazemipour/DIAYN-PyTorch#2]
Could you provide the source code (e.g., meta-controller.) to implement hierarchical Rl ? Based on the pretrained skills, how do we train the meta-controller to complete the complex tasks?

**Comment by akazemipour** [comment]
Hello @kevin-xuan,
Thank you for your comment!

I don't quite understand your question, could you please provide some clarity over the relevancy between my DIAYN's code and the Hierarchical RL code you expect?

As far as I guessed, you may have wanted to pre-train you agent via DIAYN to take advantage of its pre-training parameters for your downstream task, am I right?
If that's the case, then the source code of my DIAYN implementation is already open and released in [this repo](https://github.com/alirezakazemipour/DIAYN-PyTorch). You may find other inspirational implementations in the [acknowledge](https://github.com/alirezakazemipour/DIAYN-PyTorch#acknowledgment) section as well.

Simply train your agent by DIAYN and choose the skill you are interested in the most among the ones that your agent has learned and use that skill to boost your downstream learning task.

**Comment by kevin-xuan** [comment]
In Question 6 of Section 4.2.2,  you said that "To use the discovered skills for hierarchical RL, we learn a meta-controller whose actions are to choose which skill to execute for the next k steps", but I cannot find the corresponding meta-controller code in this repo.

**Comment by akazemipour** [comment]
Oh! There is a bit of confusion here!
I'm not among the original authors of DIAYN paper so you should ask them as, it is their idea and they have the answer of what you are looking for.

I have simply reproduced the core concept introduced in DIAYN so, those complementing parts were out of my intentions and you can not find anything relating to those parts.

Please reach out to the great [Ben Eysenbach](https://github.com/ben-eysenbach/sac). He will give you better explanations!

**Comment by kevin-xuan** [comment]
Thank you for your response!
